src/straPy/bootstrap.py

```diff
@@ -82,27 +83,72 @@ def bootstrap_distribution(sample, rep, n="auto", estimator="mean", random_seed=
         axis=1
     )
 
-def calculate_boot_ci(dist, level=0.95):
-    """Calculates a confidence interval for a distribution.
+def calculate_boot_stats(sample, rep, n="auto", level=0.95, estimator="mean", random_seed=None):
+    """Calculates a bootstrapped confidence interval for a sample.
 
-    A confidence interval for the provided sampling distribution
-    `dist` is calculated for a confidence level `level`.
+    A bootstrapped confidence interval for the desired estimator for
+    the provided sample is calculated for a confidence level `level`.
+    The mean and standard deviation of the sample are also returned.
 
     Parameters
     ----------
-    dist : numpy.ndarray
-        bootstrapped sampling distribution
+    sample : list or numpy.ndarray or pandas.core.series.Series 
+        sample to bootstrap
+    rep : int
+        number of replicates of the distribution
+    n : str or int, default="auto"
+        bootstrap sample size, "auto" specifies using the same size as the sample
     level : float, default=0.95
         confidence level
+    estimator : {"mean", "median", "var", "sd"}
+        sampling distributor's estimator
+    random_seed : None or int, default=None
+        seed for random state
     
     Returns
     -------
-    numpy.ndarray
-        lower and upper bounds of the confidence interval
+    dictionary
+        Dictionary containing lower and upper bootstrapped confidence
+        interval for the desired estimator, along with the given estimator,
+        mean, and sd of the original sample.
     
     Examples
     --------
-    >>> calculate_boot_ci([1, 1, 2, 3, 5, 10])
-    array([1., 9.375])
+    >>> calculate_boot_stats([1, 2, 3, 4], 1000, level=0.95, random_seed=123)  
+    {'lower': 1.5, 'upper': 3.5, 'sample_mean': 2.5, 'sample_sd': 1.118033988749895}
     """
-    pass
\ No newline at end of file
+
+    if not isinstance(level, float):
+        raise TypeError("level should be of type 'float")
+
+    if not (level > 0 and level < 1):
+        raise ValueError("level should be between 0 and 1")
+
+    if level < 0.7:
+        warnings.warn("Warning: chosen level is quite low--level is a confidence level, not a signficance level")
+
+    # get the bootstrapped mean vector
+    supported_estimators = {
+        "mean": np.mean,
+        "median": np.median,
+        "var": np.var,
+        "sd": np.std
+    }
```

Comment:
@BooleanJulien I think we both use this dictionary, instead of recreating it again we can refactor it into a function called `get_supported_estimators()`

Reply:
Added this function